spectractor/tools.py

```diff
@@ -2469,6 +2472,50 @@ def uvspec_available():
     return get_uvspec_binary() is not None
 
 
+def test_connectivity():
+    """Test getting all necessary external files for running the tests.
+
+    Any calspec standards which require data should be added to the list.
+
+    Because the files can move/be unavailable, and/or the remote server might
+    just be inaccessible for some reason, we don't want to fail the tests if
+    this happens.
+
+    Note that this will also populate the file in the cache if it succeeds, so
+    even if the server is unavailable by the time the test codes goes to the
+    file, it should take the cached copy, and the test should pass.
+
+    Returns
+    -------
+    success : `bool`
+        Were the necessary file(s) successfully obtained?
+    """
+
+    def patchSimbadURL(simbad):
+        """Monkeypatch the URL that Simbad is using to force it to use https.
+        """
+        simbad.SIMBAD_URL = (simbad.SIMBAD_URL.replace('http', 'https')
+                             if 'https' not in simbad.SIMBAD_URL else simbad.SIMBAD_URL)
+
+    for target in ['HD 111980']:
+        try:
+            calspec = getCalspec.Calspec(target)
+            calspec.get_spectrum_numpy()
+        except (RuntimeError, requests.exceptions.HTTPError):
+            return False
+
+        try:
+            simbadQuerier = SimbadClass()
```

Comment:
Do these two lines need to be inside the loop?

Reply:
I think the `astroquery_label` comes from `calspec` variable which depends on target.